tests/rpm/test_rpm.py

```diff
@@ -0,0 +1,93 @@
+from tests.product.standalone.presto_installer import StandalonePrestoInstaller
+
+
+class TestRpm(BaseProductTestCase):
+    def setUp(self):
+
+        super(TestRpm, self).setUp()
+        self.setup_cluster(NoHadoopBareImageProvider(), self.PA_ONLY_CLUSTER)
+
+    @docker_only
+    def test_install_fails_no_perm_to_local_path(self):
+        installer = StandalonePrestoInstaller(self)
+        rpm_name = installer.copy_presto_rpm_to_master()
+        self.upload_topology()
+        self.run_prestoadmin('configuration deploy')
+
+        script = 'chmod 600 /mnt/presto-admin/%(rpm)s; su app-admin -c ' \
+                 '"./presto-admin server install /mnt/presto-admin/%(rpm)s "'
+        error_msg = '\nFatal error: [%(host)s] error: ' \
+                    '/mnt/presto-admin/%(rpm)s: ' \
+                    'open failed: Permission denied\n\nAborting.\n'
+        expected = ''
+        for host in self.cluster.all_internal_hosts():
+            expected += error_msg % {'host': host, 'rpm': rpm_name}
+        actual = self.run_script_from_prestoadmin_dir(script, rpm=rpm_name,
+                                                      raise_error=False)
+        self.assertEqualIgnoringOrder(actual, expected)
+
+    @docker_only
+    def test_install_fails_java8_not_found(self):
+        installer = StandalonePrestoInstaller(self)
+        for container in self.cluster.all_hosts():
+            self.cluster.exec_cmd_on_host(container,
+                                          'mv /usr/java/jdk1.8.0_40 /usr/')
+        self.upload_topology()
+        cmd_output = installer.install(pa_raise_error=False)
+        actual = cmd_output.splitlines()
+        num_failures = 0
+        for line in enumerate(actual):
+            if str(line).find('Error: Required Java version'
+                              ' could not be found') != -1:
+                num_failures += 1
+
+        self.assertEqual(4, num_failures)
+
+        for container in self.cluster.all_hosts():
+            installer.assert_uninstalled(container)
+
+    @docker_only
+    def test_server_starts_java8_in_bin_java(self):
+        installer = StandalonePrestoInstaller(self)
+        for container in self.cluster.all_hosts():
+            self.cluster.exec_cmd_on_host(container,
+                                          'mv /usr/java/jdk1.8.0_40 /usr')
+            self.cluster.exec_cmd_on_host(container,
+                                          'ln -s /usr/jdk1.8.0_40/bin/java '
+                                          '/bin/java')
+        self.upload_topology()
+
+        installer.install()
+
+        # starts successfully with java8_home set
+        output = self.run_prestoadmin('server start')
+        self.assertFalse(
+            'Warning: No value found for JAVA8_HOME. Default Java will be '
+            'used.' in output)
+
+    def test_server_starts_no_java8_variable(self):
+        self.setup_cluster(NoHadoopBareImageProvider(),
```

Comment:
If JAVA_HOME is not Java 8, this scenario would fail. At least for the cluster installation, that makes this test too brittle. Maybe one test with and without JAVA_HOME set to a Java 8 path?